opennms-dao/src/test/java/org/opennms/netmgt/dao/AlarmDaoIT.java

```diff
@@ -321,4 +323,18 @@ public void testSortOnNodeLabel() {
 		}));
 		m_alarmDao.findMatching(criteria);
 	}
+
+	/**
+	 * @see https://issues.opennms.org/browse/NMS-9480
+	 */
+	@Test
+	public void testParameterizedSql() {
+		CriteriaBuilder cb = new CriteriaBuilder(OnmsAlarm.class);
+		cb.sql("{alias}.alarmid in (?)", 1, Type.INTEGER);
+		m_alarmDao.findMatching(cb.toCriteria());
```

Comment:
Can we verify the result set as well? Right now all we know is that no exceptions were thrown.